### PR TITLE
Rails-React App

### PR DESCRIPTION
**Pull request by digia3d**
>This is a project where we implemented a connection between a Ruby on Rails back-end and React front-end. 
This is a link for testing the app: http://localhost:3000/api/v1/greetings
And a screenshot of the data that is showing in the browser (To see different greetings you need to refresh the page!)

hello-rails-rect

The requirements of the project were:

> Create a new Rails app called 'hello-rails-react' with the command rails new -j webpack my_new_rails_app.

> Verify that the bundling-rails gem is added to your Gemfile.

> Initialize your project with Git.

> Install needed node dependencies as described in the previous lesson's tutorial.

> Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.

> Update webpack.config.js to include the babel-loader as described in the previous lesson's tutorial.

> Make sure that your project has a Postgres database set up.

> Create a table for storing your messages and create 5 different greetings.

> Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).

> Create a static view that will be the root of your app.

> Create the App component with react-router and render it in your static view.

> Create the Greeting component that will display a greeting. Set it up as a route in your App component.

> Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.

> Display the random greeting in your Greeting component.

### CODE REVIEW

File: app/javascript/components/greeting.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  useEffect(() => {
+    dispatch(fetchGreeting());
+  }, [dispatch]);
+
+  return (
+    <>
+      <h1>If you want to see diferent greeting, than refresh the page 😊.</h1>
+      <ul>
+        {chooseGreeting.map((greeting) => (
```
Comment (SajjadAhmad14) [inline app/javascript/components/greeting.js]: This component should render a random greeting on the screen. The app shows a blank page. Kindly make sure the app works as expected

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Alex, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/javascript/components/greeting.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    dispatch(fetchGreeting());
+  }, [dispatch]);
+
+  return (
+    <div>
+      <h1>If you want to see diferent greeting, than refresh the page 😊.</h1>
+      <ul>
+        {greeting.map((greeting) => (
+          <li key={greeting.id}>{greeting.greeting}</li>
+        ))}
```
Comment (Davidosky007) [inline app/javascript/components/greeting.js:19]: * As pointed out by the previous reviewer your API works fine only it is not connected to your react properly therefore is doesn't show any random greetings Kindly fix this 🙏

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @digia3d 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* No linter errors 👍 
* Well written PR description 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
* I strongly suggest you update your readme setup section with the proper steps you use that works for you at your end   
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/components/greeting.js (hunk 2 above)
Comment (digia3d) [reply app/javascript/components/greeting.js:19]: @Davidosky007 
I tried everything I know, but nothing worked. Can you please help with that?

**Pull request closed without merging**